### PR TITLE
Make growth deterministic with no nearby enemies

### PR DESCRIPTION
**Pull request by smitnich**
Currently players can get any order of plant priority when using growth
by repeatedly activating and cancelling the ability for no cost, so
make growth always use the same order when no monsters are nearby.

### CODE REVIEW

**Comment by neilmoore** [comment]
In the wrath code, the one-argument shuffle_array(radius_points[0]); should work just fine.  No harm either way (it expands to exactly the same call you are already making), just less typing.

File: crawl-ref/source/godwrath.cc [hunk 1]
```diff
@@ -1493,6 +1493,9 @@ static bool _fedhas_summon_plants()
     // (assuming the player isn't already surrounded).
     else if (!radius_points[0].empty())
     {
+        // Randomize the order so that plants don't always appear
+        // in the same places when no enemies are nearby
+        shuffle_array(radius_points);
```
Comment (wheals) [inline crawl-ref/source/godwrath.cc]: Is this correct, or should it be `radius_points[0]`?

Comment (neilmoore) [inline crawl-ref/source/godwrath.cc]: Should still be radius_points[0], yes.

**Comment by reaverb-Crawl** [comment]
I've implemented a different fix as of 0.17-a0-2209-g7c46689, not letting the player abort using the Fedhas growth ability.

A couple awkward problems with this solution that pushed me away from it:

A) This makes different directions on the game gird mechanically different. An example where this could be relevant is if you find a sleeping unique, retreat, and then want to create a plant out of their LOS to completely block them off.

B) The Fedhas growth targeter reveals the locations of adjacent invisible monsters. This is arguably a bug, but using the abort solution also solves this.

**Pull request closed without merging**